### ISSUE TITLE
Newer kernel version

### ISSUE THREAD

**Issue by bojidartonchev** [radxa/meta-radxa#28]
Hello, 

Are there any plans supporting newer kernel and mesa version for Rock Pi 4 models. Currently meta-radxa is using downstream linux 4.4 and mesa 20.0.2, which is 2 years old.

Best regards!

**Comment by a1batross** [comment]
You can just compile mainline Linux, it works perfectly.

**Comment by bojidartonchev** [comment]
@a1batross I guess you mean [this](https://git.yoctoproject.org/meta-rockchip). I tried and it works well. But there is some issue while booting and psplash is not starting.

`root@rock-pi-4b:~# psplash[175]: Error opening /dev/fb0`

Do you have the same problem?

Best regards!

**Comment by a1batross** [comment]
@bojidartonchev no, I skipped this repo entirely (though it can be a good start for compiling mainline Yocto).

I just added a recipe for mainline Linux, deployed it, and it works. Not LTS, just the latest 5.19.

Oh, Radxa's shitty U-Boot code needs overlay and hw_int.conf stuff, just copypaste overlays folder from old Linux tree to new and add a rule in Makefile to build it.

**Comment by bojidartonchev** [comment]
@a1batross If I understand correct, you used this meta-radxa with extra .bb for the mainline kernel.

I guess after that I have to change these lines in the machine config?

```
PREFERRED_PROVIDER_virtual/kernel = "linux-rockpi-4"
PREFERRED_VERSION_linux-rockchip = "4.4"
```

Could you please share that recipe?

Best regards!
